### PR TITLE
[JENKINS-58329] Added environment variable parsing in file path for App Center Plugin

### PR DESCRIPTION
**Pull request by wjymtg**
A minor change to enable environment variable in file path. I've tested it to be working last week. The latest commit seems to make this plugin incompatible with the very old jenkins version I'm using. So I haven't been able to test it in jenkins, but it passed all tests.

### CODE REVIEW

**Comment by mezpahlan** [comment]
Hi @wjymtg thanks for this contribution and apologies for the late reply. You're correct, we did bump the minimum version of Jenkins that this plugin supports. For now I don't think we'll lower it.

I'll take a look at this PR now and provide some comments.

File: src/main/java/io/jenkins/plugins/appcenter/AppCenterRecorder.java [hunk 1]
```diff
@@ -100,11 +101,26 @@ private boolean uploadToAppCenter(Run<?, ?> run, FilePath filePath, TaskListener
...
-                    getApiToken(), getOwnerName(), getAppName(), getPathToApp(), getBaseUrl()
-            );
-
-            return filePath.act(new UploadTask(filePath, taskListener, appCenterServiceFactory));
+        	EnvVars vars = run.getEnvironment(taskListener);
+        	String pathParsed = vars.expand(getPathToApp());
+        	logger.println("Path is " + pathParsed);
+        	FilePath localFiles[] = filePath.list(pathParsed);
+        	if(localFiles.length == 0) {
+        		logger.println("No file found to upload: " + pathParsed);
```
Comment (mezpahlan) [inline src/main/java/io/jenkins/plugins/appcenter/AppCenterRecorder.java]: 1. We do this check in `UploadTask`.
2. We want to move as much business logic as possible away from Jenkins and into the tasks themselves (easier to test).

File: src/main/java/io/jenkins/plugins/appcenter/AppCenterRecorder.java [hunk 2]
```diff
@@ -100,11 +101,26 @@ private boolean uploadToAppCenter(Run<?, ?> run, FilePath filePath, TaskListener
...
+        		return false;
+        	}
+        	boolean result = true;
+        	ArrayIterator localFilesIterator = new ArrayIterator(localFiles);
```
Comment (mezpahlan) [inline src/main/java/io/jenkins/plugins/appcenter/AppCenterRecorder.java]: I don't want to support multiple uploads in this way..... or at all to be honest. If you need to upload multiple files I recommend calling the plugin multiple times instead. In the future we need to upload obfuscation files along with the app itself. Managing a collection of potentially differing sizes isn't something I want to do.

**Comment by wjymtg** [comment]
Hi Mez,

Thanks for the comment and I totally agree. Unfortunately at the moment I'm
quite overwhelmed by my tasks at hand. I'll try update once I'm free.

**Comment by santam85** [comment]
Since I needed this urgently, I merged @wjymtg code and addressed the above comments in #19 
Please review at your earliest convenience!

**Comment by Timmi88** [comment]
Is it possible to have the same done for the App Name as well? I would need to get that one from an environment variable too.

**Comment by mezpahlan** [comment]
@Timmi88 sure it is! Could you raise a defect in Jenkins JIRA for this please. Have a look at the README for instructions. I can have a look over the weekend.

**Comment by mezpahlan** [comment]
Closing in favour of #19

**Pull request closed without merging**